process.c

```diff
@@ -619,18 +619,30 @@ proc_s_last_status(VALUE mod)
     return rb_last_status_get();
 }
 
+static VALUE
+rb_process_status_initialize(VALUE self, VALUE pid, VALUE status, VALUE error)
+{
+    struct rb_process_status *data = RTYPEDDATA_DATA(self);
+
+    data->pid = NUM2PIDT(pid);
+    data->status = NUM2INT(status);
+    data->error = NUM2INT(error);
```

Comment:
It should rb_obj_freeze(self); here

Reply:
Even in the `initialize` method? It makes sense to me for the `_new` variant, but for `_initialize`. Well,I don't have a strong opinion but this should probably just be treated like a struct?

Reply:
Yes, all Process::Status instances should be frozen, always.

Reply:
I don't agree with this. Instances returned by `Process::Status.wait` are frozen. But no objects that I'm aware of are frozen after calling `new`.

Reply:
The whole class (i.e., all current instances of that class that one can get) is immutable, there is no point to introduce a way to make it mutable, is it?